### PR TITLE
Refactor tests and yml files

### PR DESCRIPTION
**Pull request by danieljurek**
* remove old client.yml and tests.yml files from .azure-pipelines
* refactor archetype-sdk-client.yml and add a cleaner client.yml file to /eng/pipelines
* DRY up pipeline YAML
* create tests.yml for eventhub and servicebus

@mitchdenny - I've refactored the client.yml file to use archetype-sdk-client.yml. The PyPy tests use a `$(PythonVersion)` instead of building now. 

@scbedd - I've replaced `$(BuildTargetingString)` with the constant `azure-*` and we're now using `--service=`

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by danieljurek** [comment]
@Azure/azure-sdk-eng

File: eng/pipelines/templates/steps/build-test.yml [hunk 1]
```diff
@@ -9,28 +15,31 @@ steps:
...
     inputs:
       scriptPath: '$(Build.BinariesDirectory)/sdk-tools/scripts/python/verify_agent_os.py'
-      arguments: $(OSName)
+      arguments: ${{ parameters.OSName }}
 
   - script: |
       pip install pathlib twine codecov beautifulsoup4
     displayName: 'Prep Environment'
 
+  - ${{ parameters.BeforeTestSteps }}
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-test.yml]: It doesn't look like this is used by anyone. Do you have plans for people to use it? If not I would remove it for now.

Comment (danieljurek) [reply eng/pipelines/templates/steps/build-test.yml]: The plan is to use it for KeyVault. Individual tests for KeyVault can be run in parallel for considerable time savings. 

In the case of KeyVault we will use this step to pip install a package that enables us to run the parallel tests. That way we keep the parallel test work scoped just to the services that support it. If it works well we use this method in other places we may push the package up into the `Prep Environment` step and remove `BeforeTestSteps`.

**Pull request merged**